File: chapter-02-simple-md/examples/on2_neighbor/7/create_xyz.m
clear; close all;
r0 = [0.0, 0.0, 0.5, 0.5; ...
      0.0, 0.5, 0.0, 0.5; ...
      0.0, 0.5, 0.5, 0.0].';
n0 = size(r0, 1);
nxyz = 7 * [1, 1, 1];
N = nxyz(1) * nxyz(2) * nxyz(3) * n0;
a = 5.385 * [1, 1, 1];
box_length = diag(a .* nxyz);

r = zeros(N, 3);
n = 0;
for nx = 0 : nxyz(1) - 1
    for ny = 0 : nxyz(2) - 1
        for nz = 0 : nxyz(3) - 1
            for m = 1 : n0
                n = n + 1;
                r(n, :) = a .* ([nx,ny,nz] + r0(m, :));   
            end
        end
    end
end

fid = fopen('xyz.in', 'w');
fprintf(fid, '%d\n', N);
fprintf(fid, '%g %g %g %g %g %g %g %g %g\n', box_length);
for n =1 : N
    fprintf(fid, 'Ar %g %g %g %g\n', r(n, :), 40);
end
fclose(fid);

File: chapter-06-simple_md/src/ex1/create_xyz.py
import numpy as np

r0 = np.array([[0.0, 0.0, 0.5, 0.5],
               [0.0, 0.5, 0.0, 0.5],
               [0.0, 0.5, 0.5, 0.0]]).T

n0 = r0.shape[0]
nxyz = 6 * np.array([1, 1, 1])
N = nxyz[0] * nxyz[1] * nxyz[2] * n0
a = 5.385 * np.array([1, 1, 1])
box_length = a * nxyz

r = np.zeros((N, 3))
n = 0

for nx in range(nxyz[0]):
    for ny in range(nxyz[1]):
        for nz in range(nxyz[2]):
            for m in range(n0):
                n += 1
                r[n-1, :] = a * (np.array([nx, ny, nz]) + r0[m, :])

with open('xyz.in', 'w') as fid:
    fid.write(f'{N}\n')
    fid.write(f'{box_length[0]} {box_length[1]} {box_length[2]}\n')
    for n in range(N):
        fid.write(f'Ar {r[n, 0]} {r[n, 1]} {r[n, 2]} 40\n')
